SmartInsiderIntentionUniverse.cs

```diff
@@ -0,0 +1,117 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using Newtonsoft.Json;
+using System;
+using System.Globalization;
+using System.IO;
+using QuantConnect.Data;
+using QuantConnect.Logging;
+
+namespace QuantConnect.DataSource
+{
+    /// <summary>
+    /// Smart Insider Intentions Universe
+    /// </summary>
+    public class SmartInsiderIntentionUniverse : BaseData
+    {
+        private TimeSpan _period = TimeSpan.FromDays(1);
+        
+        /// <summary>
+        /// Number of shares to be or authorised to be traded
+        /// </summary>
+        public int? Amount { get; set; }
+
+        /// <summary>
+        /// Value of shares to be authorised to be traded
+        /// </summary>
+        public long? AmountValue { get; set; }
+
+        /// <summary>
+        /// Percentage of oustanding shares to be authorised to be traded
+        /// </summary>
+        public decimal? Percentage { get; set; }
+
+        /// <summary>
+        /// Minimum price shares will or may be purchased at
+        /// </summary>
+        public decimal? MinimumPrice { get; set; }
+
+        /// <summary>
+        /// Maximum price shares will or may be purchased at
+        /// </summary>
+        public decimal? MaximumPrice { get; set; }
+
+        /// <summary>
+        /// Market Capitalization in USD
+        /// </summary>
+        public decimal? USDMarketCap { get; set; }
```

Comment:
By the way the universe classes are created they wont have null values? same for `SmartInsiderTransactionUniverse`

Reply:
They fetch together with the base `SmartInsiderTransaction` class, so if the raw data dun have data, this should also has no data?

Reply:
hm right, so I think the converter might blow up in that case? it's not taking into account possible empty values? Instead of having to keep a string and parse it it would keep a reference to the universe object and aggregate on it maybe?
![image](https://user-images.githubusercontent.com/18473240/163897574-afaa6ec9-1126-4901-9dc4-1bae1be82d76.png)